### PR TITLE
feat: Implement Logger

### PR DESCRIPTION
**Pull request by sang-w0o**
# Implement Logger

- This PR implements Logger, which logs certain messages depending on the log level configured.   
  Log level contains of:
  - Info: Default log level, only logs informational messages.
  - Debug: Logs all incoming requests, and responses to help debugging process.
  - Off: Doesn't log anything at all.

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	// Debug helps you debug mini-memcached usages,
+	//and logs all command and its result after being processed.
+	Debug LogLevel = "debug"
+	// Off does not log anything.
+	Off LogLevel = "off"
+)
+
+// Logger logs certain messages depending on the LogLevel.
+// TODO:(@sang-w0o): Selectively accept where log output will go to.
+type Logger struct {
```
Comment (HurSungYun) [inline logger.go:23]: I believe Logger should be implemented as an interface.

Implementation in GORM can be a good reference. https://github.com/go-gorm/gorm/blob/master/logger/logger.go

I would suggest interface below.

```go
// Logger logger interface
type Logger interface {
	Info(string)
	Debug(string)
}
```

Furthermore,

1. Checking log level each time is inefficient
2. log levels are cascading. (e.g. INFO level logs should be printed in DEBUG log level)

File: minimemcached.go [hunk 2]
```diff
@@ -54,12 +56,13 @@ type item struct {
 
 type Option func(m *MiniMemcached)
 
-// newMiniMemcached returns a newMiniMemcached, non-started, MiniMemcached object.
-func newMiniMemcached(opts ...Option) *MiniMemcached {
+// newMiniMemcached returns a non-started, MiniMemcached object.
+func newMiniMemcached(lv LogLevel, opts ...Option) *MiniMemcached {
```
Comment (HurSungYun) [inline minimemcached.go:60]: it would be better to use Options pattern for log level.

**Review by HurSungYun — CHANGES_REQUESTED** [review]
I believe Logger should be declared as an Interface and minimemcached should support DefaultLogger as default implementation.

Please make Logger as an interface and convert existing logger implementation into `type DefaultLogger struct` or something.

**Comment by sang-w0o** [comment]
I will try to update this PR accepting @HurSungYun 's recommendations soon!

**Comment by novemberde** [comment]
Please reopen the pull request as it is no longer being followed up

**Pull request closed without merging**